handler.go

```diff
@@ -343,15 +278,25 @@ func (s *handler) createError(err error) *respError {
 		}
 	}
 
-	out := &respError{
+	out := &JSONRPCError{
 		Code:    code,
 		Message: err.Error(),
 	}
 
-	if m, ok := err.(marshalable); ok {
-		meta, err := m.MarshalJSON()
-		if err == nil {
+	switch m := err.(type) {
+	case RPCErrorCodec:
+		o, err := m.ToJSONRPCError()
+		if err != nil {
+			log.Warnf("Failed to convert error to JSONRPCError: %v", err)
```

Comment:
let's `Errorf` these two as well, if they're not converting then it's either a programmer error or a system error (like OOM)

Reply:
Done.